app/src/main/res/xml/preferences_master.xml

```diff
@@ -350,6 +350,12 @@
                 android:key="@string/pref_key__editor_enable_line_breaking"
                 android:summary="@string/soft_break_if_not_fit_to_width_else_enable_horizontal_scrolling"
                 android:title="@string/wrap_words" />
+            <CheckBoxPreference
+                android:defaultValue="false"
+                android:icon="@drawable/ic_line_numbers_24dp"
+                android:key="@string/pref_key__editor_line_numbers"
+                android:summary="@string/show_line_numbers_on_the_left_side_of_the_editor"
+                android:title="@string/show_line_numbers" />
```

Comment:
I suggest this time this approach on settings and on this topic:

* Should be enough to have one settings key like "show_line_numbers", which enables both line numbers at code blocks (view mode) as well at editor.
* No new option at Settings screen (=remove the 2 added blockes at prefereences_master.xml)
* Just have one isShowLineNumbers() method for checking if it is enabled. And a toggleShowLineNumbers() which toggles it globally. So one global (non-per-file-specific) setting.

Would that be OK to you? It also means the logic gets a bit easier and less 😄 .

Reply:
OK, maybe one is enough. I just think that someone may always hope to enable line numbers, so they can open it at Settings screen. another option for others just occasionally enable line numbers, they don't to need to Settings screen to switch. You can make some adjustments with your ideas 😄.